src/api/upload.ts

```diff
@@ -80,6 +81,17 @@ router.head<UploadState>("/upload", async (ctx) => {
 });
 
 router.put<UploadState>("/upload", async (ctx) => {
+  if (!config.upload.enabled) throw new HttpErrors.NotFound("Uploads disabled");
+
+  const pubkey = ctx.state.auth?.pubkey;
+  if (!pubkey) throw new HttpErrors.Unauthorized("Missing public key");
+
+  await fetchWhitelist(); // Ensure the whitelist is up-to-date
+
+  if (config.whitelist.enabled && pubkey && !isWhitelisted(pubkey)) {
```

Comment:
This code should be moved to the `checkUpload` method ( in the `"check auth" block ) so it applies to both the `/upload` and `/media` endpoint

It also wont have to check if `ctx.state.auth` exists then ( there are some configurations that allow uploads without auth )